tests/net/lib/lwm2m/lwm2m_rd_client/src/main.c

```diff
@@ -244,7 +288,36 @@ ZTEST(lwm2m_rd_client, test_start_registration_update)
 		     NULL);
 
 	lwm2m_rd_client_update();
-	zassert_true(check_lwm2m_rd_client_event(LWM2M_RD_CLIENT_EVENT_REG_UPDATE_COMPLETE, 2),
+	zassert_true(check_lwm2m_rd_client_event(LWM2M_RD_CLIENT_EVENT_REG_UPDATE_COMPLETE, 3),
+		     NULL);
+}
+
+ZTEST(lwm2m_rd_client, test_rx_off)
+{
+	struct lwm2m_ctx ctx;
+
+	(void)memset(&ctx, 0x0, sizeof(ctx));
+
+	test_prepare_pending_message_cb(&message_reply_cb_default);
+
+	lwm2m_engine_add_service_fake.custom_fake = lwm2m_engine_add_service_fake_default;
+	lwm2m_rd_client_init();
+	test_lwm2m_engine_start_service();
+	wait_for_service(1);
+
+	lwm2m_get_bool_fake.custom_fake = lwm2m_get_bool_fake_default;
+	lwm2m_sprint_ip_addr_fake.custom_fake = lwm2m_sprint_ip_addr_fake_default;
+	lwm2m_init_message_fake.custom_fake = lwm2m_init_message_fake_default;
+	coap_header_get_code_fake.custom_fake = coap_header_get_code_fake_created;
+	coap_find_options_fake.custom_fake = coap_find_options_do_registration_reply_cb_ok;
+	zassert_true(lwm2m_rd_client_start(&ctx, "Test", 0, lwm2m_event_cb, lwm2m_observe_cb) == 0,
+		     NULL);
+	zassert_true(check_lwm2m_rd_client_event(LWM2M_RD_CLIENT_EVENT_REGISTRATION_COMPLETE, 0),
+		     NULL);
+
+	engine_update_tx_time();
+	k_sleep(K_SECONDS(15));
```

Comment:
Do we really need such a long sleep here, seems quite excessive

Reply:
This test runs on `native_posix` target and does not require "-rt" flag, so it is not effectively 15 seconds in CI.

Example on my machine
```
------ TESTSUITE SUMMARY START ------

SUITE PASS - 100.00% [lwm2m_rd_client]: pass = 13, fail = 0, skip = 0, total = 13 duration = 56.610 seconds
...
------ TESTSUITE SUMMARY END ------

===================================================================
PROJECT EXECUTION SUCCESSFUL
west build -t run  0,22s user 0,03s system 104% cpu 0,243 total
```

But yes, we need a longer sleep here, because in settings 10 second is the RX window, and only after that we receive the RX_OFF event.